monai/transforms/utility/array.py

```diff
@@ -1715,3 +1715,126 @@ def __call__(self, img: NdarrayOrTensor, meta_dict: Mapping | None = None) -> Nd
+        affine_lps_to_ras: Defaults to ``False``. Set this to ``True`` if all of the following are true:
+            1) The image is read by `ITKReader`,
+            2) The `ITKReader` has `affine_lps_to_ras=True`,
+            3) The data is in world coordinates.
+            This ensures the correct application of the affine transformation between LPS (left-posterior-superior)
+            and RAS (right-anterior-superior) coordinate systems. This argument ensures the points and the affine
+            matrix are in the same coordinate system.
+    """
+
+    def __init__(
+        self,
+        dtype: DtypeLike | torch.dtype = torch.float64,
+        affine: torch.Tensor | None = None,
+        invert_affine: bool = True,
+        affine_lps_to_ras: bool = False,
+    ) -> None:
+        self.dtype = dtype
+        self.affine = affine
+        self.invert_affine = invert_affine
+        self.affine_lps_to_ras = affine_lps_to_ras
+
+    @staticmethod
+    def apply_affine_to_points(data: torch.Tensor, affine: torch.Tensor, dtype: DtypeLike | torch.dtype):
+        data_: torch.Tensor = convert_to_tensor(data, track_meta=False, dtype=torch.float64)
+        affine = to_affine_nd(data_.shape[-1], affine)
+
+        homogeneous = concatenate((data_, torch.ones((data_.shape[0], data_.shape[1], 1))), axis=2)
+        transformed_homogeneous = torch.matmul(affine, homogeneous.transpose(1, 2))
+        transformed_coordinates = transformed_homogeneous[:, :-1, :].transpose(1, 2)
+        out, *_ = convert_to_dst_type(transformed_coordinates, data, dtype=dtype)
+
+        return out
+
+    def transform_coordinates(self, data: torch.Tensor, affine: torch.Tensor | None = None):
+        """
+        Transform coordinates using an affine transformation matrix.
+
+        Args:
+            data: The input coordinates, assume to be in shape (C, N, 2 or 3).
+            affine: A 3x3 or 4x4 affine transformation matrix.
+
+        Returns:
+            Transformed coordinates.
+        """
+        data = convert_to_tensor(data, track_meta=get_track_meta())
+        applied_affine = data.affine if isinstance(data, MetaTensor) else None
+
+        if affine is None and self.invert_affine:
+            raise ValueError("affine must be provided when invert_affine is True.")
+
+        affine = applied_affine if affine is None else affine
+        affine = convert_data_type(affine, dtype=torch.float64)[0]  # always convert to float64 for affine
+        original_affine = affine
+        if self.affine_lps_to_ras:
+            affine = orientation_ras_lps(affine)
+
+        affine_inv = linalg_inv(affine)
+
+        if self.invert_affine:
+            _affine = affine_inv
+            # consider the affine transformation already applied to the data in the world space
+            if applied_affine is not None:
+                _affine = _affine @ linalg_inv(applied_affine)
+        else:
+            _affine = affine
```

Comment:
```suggestion
        _affine = affine
        if self.invert_affine:
            _affine = linalg_inv(affine)
            # consider the affine transformation already applied to the data in the world space
            if applied_affine is not None:
                _affine = _affine @ linalg_inv(applied_affine)            
```

I remember that some checkers have issue with variables declared in `if-else` blocks and not beforehand, these are erroneously flagged as being non-existent.

Reply:
Thanks, updated it.